src/contrib/persistence/Akka.Persistence.Sqlite/Snapshot/ConflictNameResolver.cs

```diff
@@ -0,0 +1,53 @@
+// //-----------------------------------------------------------------------
+// // <copyright file="ConflictNameResolver.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2019 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2019 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.Sqlite;
+
+namespace Akka.Persistence.Sqlite.Snapshot
+{
+    /// <summary>
+    /// Check if there is a conflict with snapshot table name
+    /// </summary>
+    internal class ConflictNameResolver
+    {
+        /// <summary>
+        /// Tries to resolve the conflict of snapshot table names
+        /// </summary>
+        /// <param name="connection">sqlite connection</param>
+        /// <param name="configuration">sqlite query configuration</param>
+        /// <returns>Returns true if has the conflict</returns>
+        public async Task<bool> Resolve(SqliteConnection connection, SqliteQueryConfiguration configuration)
```

Comment:
... Why is this class necessary? I don't understand the need for the rest of the changes in this PR. I thought it was just a HOCON fallback?

Reply:
Did not get the point with HOCON fallback :)
The idea was to do not get db corrupted for people who upgrade akka library.
This fix gets akka still use old hardcoded table 'snapshot' despite of 'table-name' setting value.

Reply:
I'll take another look at it.